riak_test/multiple_dcs_node_failure_test.erl

```diff
@@ -0,0 +1,350 @@
+    rt:update_app_config(all,[
+        {riak_core, [{ring_creation_size, NumVNodes}]}
+    ]),
+
+    [Cluster1, Cluster2, Cluster3] = rt:build_clusters([1,1,1]),
+    rt:wait_until_ring_converged(Cluster1),
+    rt:wait_until_ring_converged(Cluster2),
+    rt:wait_until_ring_converged(Cluster3),
+
+    {ok, Prot} = rpc:call(hd(Cluster1), application, get_env, [antidote, txn_prot]),
+    ?assertMatch(clocksi, Prot),
+
+    ok = common:setup_dc_manager([Cluster1, Cluster2, Cluster3], first_run),
+    cluster_failure_test(Cluster1, Cluster2, Cluster3),
+
+    ok = common:just_clean_clusters([Cluster1, Cluster2, Cluster3]),
+    [Cluster1, Cluster2, Cluster3] = rt:build_clusters([1,1,1]),
+    rt:wait_until_ring_converged(Cluster1),
+    rt:wait_until_ring_converged(Cluster2),
+    rt:wait_until_ring_converged(Cluster3),
+
+    ok = common:setup_dc_manager([Cluster1, Cluster2, Cluster3], first_run),
+    update_during_cluster_failure_test(Cluster1, Cluster2, Cluster3),
+
+    ok = common:just_clean_clusters([Cluster1, Cluster2, Cluster3]),
+    [Cluster1a, Cluster2a] = rt:build_clusters([2,1]),
+    rt:wait_until_ring_converged(Cluster1a),
+    rt:wait_until_ring_converged(Cluster2a),
+
+    ok = common:setup_dc_manager([Cluster1a, Cluster2a], first_run),
+    multiple_cluster_failure_test(Cluster1a, Cluster2a),
+
+    pass.
+
+%% In this test there are 3 DCs each with 1 node
+%% The test starts by performing some updates, ensuring they are propagated
+%% The it kills the node of the first DC
+%% Once restarted it checks that updates are still performed safely
+%% and propagated to other DCs
+cluster_failure_test(Cluster1, Cluster2, Cluster3) ->
+    Node1 = hd(Cluster1),
+    Node2 = hd(Cluster2),
+    Node3 = hd(Cluster3),
+
+    Key1 = cluster_failure_test,
+    
+    WriteResult1 = rpc:call(Node1,
+                            antidote, append,
+                            [Key1, riak_dt_gcounter, {increment, ucl1}]),
+    ?assertMatch({ok, _}, WriteResult1),
+    WriteResult2 = rpc:call(Node1,
+                            antidote, append,
+                            [Key1, riak_dt_gcounter, {increment, ucl2}]),
+    ?assertMatch({ok, _}, WriteResult2),
+    WriteResult3 = rpc:call(Node1,
+                            antidote, append,
+                            [Key1, riak_dt_gcounter, {increment, ucl3}]),
+    ?assertMatch({ok, _}, WriteResult3),
+    {ok,{_,_,CommitTime}}=WriteResult3,
+    ReadResult = rpc:call(Node1, antidote, read,
+                          [Key1, riak_dt_gcounter]),
+    ?assertEqual({ok, 3}, ReadResult),
+
+    %% Kill and restart a node an be sure everyhing works
+    lager:info("Killing and restarting node ~w", [Node1]),
+    timer:sleep(5000),
+    rt:brutal_kill(Node1),
+    rt:start_and_wait(Node1),
+    rt:wait_until_ring_converged(Cluster1),
+    lager:info("Waiting until vnodes are restarted"),
+    rt:wait_until(Node1, fun wait_init:check_ready/1),
+    timer:sleep(10000),
```

Comment:
It's to be sure the updates have been stored to log durably.
Right now the disk log sync option is hardcoded in antidote.hrl, so maybe I should put that as one of the config options here: https://github.com/SyncFree/antidote/blob/master/src/antidote.app.src#L15
so then I can have the sync log be enabled for this test?